.github/workflows/test.yml

```diff
@@ -6,23 +6,29 @@ on:
       - develop
   pull_request: {}
 
+env:
+  HOLOCHAIN_REPO: "/var/tmp/holochain_repo"
+  CACHIX_REV: "v1.2"
+
 jobs:
   test:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
+        os: [ubuntu-latest, macos-latest]
         script: ["./test.sh", "./bench.sh"]
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: |
-            /nix/store
-          key: nix-store-${{ hashFiles('default.nix') }}-${{ hashFiles('config.nix') }}-${{ hashFiles('nix') }}
-      - uses: cachix/install-nix-action@v12
+      - uses: actions/checkout@v3
+      - name: Install nix
+        uses: cachix/install-nix-action@v20
+      - name: Setup cachix
+        uses: cachix/cachix-action@v12
+        if: ${{ ! contains(matrix.platform.runs-on, 'self-hosted') }}
         with:
-          nix_path: nixpkgs=channel:nixos-unstable
-          extra_nix_config: |
-            substituters = https://cache.nixos.org/ https://cache.holo.host/
-            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= cache.holo.host-1:lNXIXtJgS9Iuw4Cu6X0HINLu9sTfcjEntnrgwMQIMcE= cache.holo.host-2:ZJCkX3AUYZ8soxTLfTb60g+F3MkWD7hkH9y8CgqwhDQ=
-      - run: nix-shell --fallback --run ${{ matrix.script }}
+          name: holochain-ci
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          installCommand: |
+            nix-env -if https://github.com/cachix/cachix/tarball/${CACHIX_REV} \
```

Comment:
@steveej do you know what this check is for? Is it still necessary? We do it in Holochain but I don't know why

Reply:
we needed a bugfix in cachix before it came fully trough their release pipeline. can be removed now.